pkg/awsutils/awsutils.go

```diff
@@ -411,7 +416,7 @@ func New(useCustomNetworking, disableENIProvisioning, v4Enabled, v6Enabled bool)
 	}
 
 	// Clean up leaked ENIs in the background
-	if !disableENIProvisioning {
+	if !disableENIProvisioning && !disableLeakedENICollection {
```

Comment:
Thinking about it..Why don't we disable this for v6 completely since v6 support in PD mode will not be creating any additional ENIs? We only have/need Primary ENI in v6 PD mode..

Reply:
Yeah we can. Only catch is if they moved from v4 to v6 cluster and there are leaked ENIs. Maybe should be fine?

Reply:
So, v6 cluster will have to be a brand new cluster and more importantly the proposed IPv6 IAM policy for an IPv6 cluster will not have permissions to `delete` an ENI, so it is not going to be of any use.

Reply:
Sure, we might but we can probably remove the v6 check at that point along with v6 specific checks we placed around custom nw code. Problem I see with this right now is the `ipamd` logs might be filled with 403s(`Unauthorized`) every hr when the `aws-node` pod attempts to delete a leaked ENI.

Reply:
Yeah makes sense to disable it for now until we have actual support for custom networking